crates/eips/src/eip4844/sidecar.rs

```diff
@@ -32,6 +36,107 @@ pub struct BlobTransactionSidecar {
     pub proofs: Vec<Bytes48>,
 }
 
+impl IntoIterator for BlobTransactionSidecar {
+    type Item = BlobTransactionSidecarItem;
+    type IntoIter = alloc::vec::IntoIter<BlobTransactionSidecarItem>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        self.blobs
+            .into_iter()
+            .zip(self.commitments)
+            .zip(self.proofs)
+            .map(|((blob, commitment), proof)| BlobTransactionSidecarItem {
+                index: 0,
+                blob,
+                kzg_commitment: commitment,
+                kzg_proof: proof,
+            })
+            .collect::<Vec<_>>()
```

Comment:
this collect is a bit pointless and could be avoided with a custom iterator type that is generic over I: Iterator (blob,commitment, proof) and uses `Enumerate<I>`

Reply:
can open a good first for this